### ISSUE TITLE
Error: Failed to decode resource from state

### ISSUE THREAD

**Issue by fernandoiury** [grafana/terraform-provider-grafana#400]
### Terraform Version
* Terraform: 1.2
* Terraform Grafana Provider: 1.20.0
* Grafana: 1.8.4

### Affected Resource(s)
Please list the resources as a list, for example:
* grafana_datasource

Getting this error after upgrading the provider to `1.20`. 1.19 is working fine. I guess probably related to https://github.com/grafana/terraform-provider-grafana/pull/222

```
╷
│ Error: Failed to decode resource from state
│ 
│ Error decoding "module.grafana.grafana_data_source.stackdriver[0]" from previous state: a number is required
╵
╷
│ Error: a number is required
│ 
│   with module.grafana.grafana_data_source.stackdriver[0],
│   on ../../modules/grafana/main.tf line 53, in resource "grafana_data_source" "stackdriver":
│   53:   json_data {
```

**Comment by julienduchesne** [comment]
It's a migration issue from https://github.com/grafana/terraform-provider-grafana/pull/398 actually. I'll try to put out a fix version ASAP

**Comment by julienduchesne** [comment]
This is fixed in 1.20.1. Let me know if you have any further issues!

**Comment by fernandoiury** [comment]
thanks @julienduchesne. It's working for me!
